### PR TITLE
Implement Series.item

### PR DESCRIPTION
**Pull request by itholic**
This PR proposes `Series.item`

```python
>>> kser = ks.Series([10])
>>> kser.item()
10
```

### CODE REVIEW

File: databricks/koalas/series.py [hunk 1]
```diff
@@ -4730,6 +4730,30 @@ def asof(self, where):
...
+        ValueError
+            If the data is not length-1.
+
+        Examples
+        --------
+        >>> kser = ks.Series([10])
+        >>> kser.item()
+        10
+        """
+        if len(self) != 1:
```
Comment (HyukjinKwon) [inline databricks/koalas/series.py:4753]: Let's avoid scanning whole data here. You can do this via collecting 2 rows.

File: databricks/koalas/series.py [hunk 2]
```diff
@@ -4730,6 +4730,30 @@ def asof(self, where):
         result_series.name = self.name
         return result_series
 
+    def item(self):
+        """
+        Return the first element of the underlying data as a python scalar.
```
Comment (HyukjinKwon) [inline databricks/koalas/series.py:4735]: python -> Python

Comment (itholic) [reply databricks/koalas/series.py:4735]: @HyukjinKwon Ah, I just followed the docstring of pandas, but It seems better to fix. Thanks!

File: databricks/koalas/series.py (hunk 1 above)
Comment (itholic) [reply databricks/koalas/series.py]: Thanks!

File: databricks/koalas/series.py [hunk 3]
```diff
@@ -4730,6 +4730,30 @@ def asof(self, where):
...
+        ValueError
+            If the data is not length-1.
+
+        Examples
+        --------
+        >>> kser = ks.Series([10])
+        >>> kser.item()
+        10
+        """
+        if len(self[:2]) != 1:
```
Comment (HyukjinKwon) [inline databricks/koalas/series.py:4753]: Can you reuse the output from `self[:2]` rather then executing the jobs twice?

Comment (itholic) [reply databricks/koalas/series.py]: Ah, I missed it. Thanks. !

File: databricks/koalas/series.py [hunk 4]
```diff
@@ -4730,6 +4730,31 @@ def asof(self, where):
...
+        Examples
+        --------
+        >>> kser = ks.Series([10])
+        >>> kser.item()
+        10
+        """
+        item_top_two = self[:2]
+        if len(item_top_two) != 1:
+            raise ValueError("can only convert an array of size 1 to a Python scalar")
+        return item_top_two[0]
```
Comment (ueshin) [inline databricks/koalas/series.py:4756]: Seems like this still runs Spark jobs twice? We should explicitly call `to_pandas()` or `collect()`?

Comment (itholic) [reply databricks/koalas/series.py:4756]: Ah, I'll check and fix it. Thanks !

Comment (HyukjinKwon) [reply databricks/koalas/series.py:4756]: Let's be very careful on this @itholic next time. This one line mistake makes the performance two times slower.

Comment (itholic) [reply databricks/koalas/series.py:4756]: @HyukjinKwon 
Yeah, I should have considered much more carefully.
Thanks for reminding me once again, @HyukjinKwon ! I must keep that in mind.

**Pull request merged**